sdk/compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -993,6 +1000,26 @@ buildEffect relativize pkgConfig@PackageConfigFields{..} opts mbOutFile incremen
             Nothing -> pure $ distDir </> name <.> "dar"
             Just out -> rel out
 
+        syncUpgradesField :: PackageConfigFields -> Options -> IO (PackageConfigFields, Options)
+        syncUpgradesField pkgConf opts = do
+          opts <- updateUpgradePath "build" opts (pUpgradeDar pkgConf)
+          pure (pkgConf { pUpgradeDar = uiUpgradedPackagePath (optUpgradeInfo opts) }, opts)
+
+updateUpgradePath :: T.Text -> Options -> Maybe FilePath -> IO Options
```

Comment:
We use this function to make sure that PackageConfigFields and Options have the same `upgradeDar`